diff -ru libtorrent-0.13.6-patched/src/torrent/peer/client_info.cc libtorrent-0.13.6-patched-old/src/torrent/peer/client_info.cc
--- libtorrent-0.13.6-patched/src/torrent/peer/client_info.cc   2020-03-10 17:43:01.000000000 +0000
+++ libtorrent-0.13.6-patched-old/src/torrent/peer/client_info.cc       2020-03-10 18:15:25.251753428 +0000
@@ -48,6 +48,7 @@
   case TYPE_TIXATI:
     return 3;
   case TYPE_AZUREUS:
+  case TYPE_CUSTOM:
     return 2;
   case TYPE_COMPACT:
   case TYPE_MAINLINE:
@@ -64,6 +65,7 @@
   case TYPE_AZUREUS:
   case TYPE_TIXATI:
     return 4;
+  case TYPE_CUSTOM:
   case TYPE_COMPACT:
   case TYPE_MAINLINE:
     return 3;
diff -ru libtorrent-0.13.6-patched/src/torrent/peer/client_info.h libtorrent-0.13.6-patched-old/src/torrent/peer/client_info.h
--- libtorrent-0.13.6-patched/src/torrent/peer/client_info.h    2020-03-10 17:43:01.000000000 +0000
+++ libtorrent-0.13.6-patched-old/src/torrent/peer/client_info.h        2020-03-10 18:19:02.338877618 +0000
@@ -49,6 +49,7 @@
     TYPE_UNKNOWN,
     TYPE_AZUREUS,
     TYPE_COMPACT,
+    TYPE_CUSTOM,
     TYPE_MAINLINE,
     TYPE_TIXATI,
     TYPE_MAX_SIZE
diff -ru libtorrent-0.13.6-patched/src/torrent/peer/client_list.cc libtorrent-0.13.6-patched-old/src/torrent/peer/client_list.cc
--- libtorrent-0.13.6-patched/src/torrent/peer/client_list.cc   2020-03-10 17:43:01.000000000 +0000
+++ libtorrent-0.13.6-patched-old/src/torrent/peer/client_list.cc       2020-03-10 18:33:50.991472733 +0000
@@ -156,6 +156,11 @@
   insert_helper(ClientInfo::TYPE_COMPACT, "O", NULL, NULL, "Osprey Permaseed");
   insert_helper(ClientInfo::TYPE_COMPACT, "R", NULL, NULL, "Tribler");
   insert_helper(ClientInfo::TYPE_COMPACT, "Q", NULL, NULL, "BTQueue");
+
+  insert_helper(ClientInfo::TYPE_CUSTOM, "BI", NULL, NULL, "BI Unknown");
+  insert_helper(ClientInfo::TYPE_CUSTOM, "BT", NULL, NULL, "Mainline");
+  insert_helper(ClientInfo::TYPE_CUSTOM, "UT", NULL, NULL, "uTorrent");
+  insert_helper(ClientInfo::TYPE_CUSTOM, "UW", NULL, NULL, "µTorrent Web");
 }
 
 ClientList::~ClientList() {
@@ -225,6 +230,17 @@
     for (int i = 0; i < 4; i++)
       dest->mutable_version()[i] = dest->mutable_upper_version()[i] = rak::hexchar_to_value(id[3 + i]);
 
+  } else if (id[0] == '-' && id[7] == '-' &&
+      std::isalpha(id[1]) && std::isalpha(id[2]) &&
+      std::isxdigit(id[3]) && std::isxdigit(id[4]) && std::isxdigit(id[5]) && std::isalpha(id[6])) {
+    dest->set_type(ClientInfo::TYPE_CUSTOM);
+
+    dest->mutable_key()[0] = id[1];
+    dest->mutable_key()[1] = id[2];
+
+    for (int i = 0; i < 3; i++)
+      dest->mutable_version()[i] = dest->mutable_upper_version()[i] = rak::hexchar_to_value(id[3 + i]);
+
   } else if (id[0] == 'T' && id[1] == 'I' && id[2] == 'X' && id[7] == '-' &&
              std::isxdigit(id[3]) && std::isxdigit(id[4]) && std::isxdigit(id[5]) && std::isxdigit(id[6])) {
     dest->set_type(ClientInfo::TYPE_TIXATI);
