diff -ur rtorrent-0.9.6/src/command_ui.cc rtorrent-0.9.6-temp-filter/src/command_ui.cc
--- rtorrent-0.9.6/src/command_ui.cc	2015-09-03 20:03:30.000000000 +0100
+++ rtorrent-0.9.6-temp-filter/src/command_ui.cc	2020-03-10 12:26:52.319056241 +0000
@@ -39,6 +39,8 @@
 #include <sys/types.h>
 
 #include <ctime>
+#include <regex>
+
 #include <rak/functional.h>
 #include <rak/functional_fun.h>
 
@@ -282,6 +284,53 @@
   return result.is_value() ? result.as_value() == 0 : (int64_t)false;
 }
 
+// Regexp based 'match' function.
+// arg1: the text to match.
+// arg2: the regexp pattern.
+// eg: match{d.name=,.*linux.*iso}
+torrent::Object apply_match(rpc::target_type target, const torrent::Object::list_type& args) {
+  if (args.size() != 2)
+    throw torrent::input_error("Wrong argument count for 'match': 2 arguments needed.");
+
+  // This really should be converted to using args flagged as
+  // commands, so that we can compare commands and statics values.
+
+  torrent::Object result1;
+  torrent::Object result2;
+
+  rpc::target_type target1 = rpc::is_target_pair(target) ? rpc::get_target_left(target) : target;
+  rpc::target_type target2 = rpc::is_target_pair(target) ? rpc::get_target_right(target) : target;
+
+  if (args.front().is_dict_key())
+    result1 = rpc::commands.call_command(args.front().as_dict_key().c_str(), args.front().as_dict_obj(), target1);
+  else
+    result1 = rpc::parse_command_single(target1, args.front().as_string());
+
+  if (args.back().is_dict_key())
+    result2 = rpc::commands.call_command(args.back().as_dict_key().c_str(), args.back().as_dict_obj(), target2);
+  else
+    result2 = args.back().as_string();
+
+  if (result1.type() != result2.type())
+    throw torrent::input_error("Type mismatch for 'match' arguments.");
+
+  std::string text = result1.as_string();
+  std::string pattern = result2.as_string();
+
+  std::transform(text.begin(), text.end(), text.begin(), ::tolower);
+  std::transform(pattern.begin(), pattern.end(), pattern.begin(), ::tolower);
+
+  bool isAMatch = false;
+  try {
+    std::regex re(pattern);
+    isAMatch = std::regex_match(text, re);
+  } catch (const std::regex_error& exc) {
+    control->core()->push_log_std("regex_error: " + std::string(exc.what()));
+  }
+
+  return isAMatch ? (int64_t)true : (int64_t)false;
+}
+
 torrent::Object
 apply_to_time(const torrent::Object& rawArgs, int flags) {
   std::tm *u;
@@ -524,6 +573,7 @@
   CMD2_ANY_LIST("view.set",           tr1::bind(&apply_view_set, tr1::placeholders::_2));
 
   CMD2_ANY_LIST("view.filter",        tr1::bind(&apply_view_event, &core::ViewManager::set_filter, tr1::placeholders::_2));
+  CMD2_ANY_LIST("view.temp_filter",   tr1::bind(&apply_view_event, &core::ViewManager::set_temp_filter, std::placeholders::_2));
   CMD2_ANY_LIST("view.filter_on",     tr1::bind(&apply_view_filter_on, tr1::placeholders::_2));
 
   CMD2_ANY_LIST("view.sort",          tr1::bind(&apply_view_sort, tr1::placeholders::_2));
@@ -549,6 +599,8 @@
   CMD2_DL        ("ui.unfocus_download",   tr1::bind(&cmd_ui_unfocus_download, tr1::placeholders::_1));
   CMD2_ANY_STRING("ui.current_view.set",   tr1::bind(&cmd_ui_set_view, tr1::placeholders::_2));
 
+  CMD2_VAR_BOOL  ("ui.console.log.tempfilter", 0);
+
   // Move.
   CMD2_ANY("print", &apply_print);
   CMD2_ANY("cat",   &apply_cat);
@@ -565,6 +617,7 @@
   CMD2_ANY_LIST("less",    &apply_less);
   CMD2_ANY_LIST("greater", &apply_greater);
   CMD2_ANY_LIST("equal",   &apply_equal);
+  CMD2_ANY_LIST("match",   &apply_match);
 
   CMD2_ANY_VALUE("convert.gm_time",      tr1::bind(&apply_to_time, tr1::placeholders::_2, 0));
   CMD2_ANY_VALUE("convert.gm_date",      tr1::bind(&apply_to_time, tr1::placeholders::_2, 0x2));
diff -ur rtorrent-0.9.6/src/core/view.cc rtorrent-0.9.6-temp-filter/src/core/view.cc
--- rtorrent-0.9.6/src/core/view.cc	2015-09-03 20:03:30.000000000 +0100
+++ rtorrent-0.9.6-temp-filter/src/core/view.cc	2020-03-10 12:22:05.704143733 +0000
@@ -89,17 +89,21 @@
 };
 
 struct view_downloads_filter : std::unary_function<Download*, bool> {
-  view_downloads_filter(const torrent::Object& cmd) : m_command(cmd) {}
+  view_downloads_filter(const torrent::Object& cmd, const torrent::Object& cmd2) : m_command(cmd), m_command2(cmd2) {}
 
   bool operator () (Download* d1) const {
-    if (m_command.is_empty())
+    return this->evalCmd(m_command, d1) && this->evalCmd(m_command2, d1);
+  }
+
+  bool evalCmd(const torrent::Object& cmd, Download* d1) const {
+    if (cmd.is_empty())
       return true;
 
     try {
       torrent::Object result;
 
-      if (m_command.is_dict_key()) {
-        // torrent::Object tmp_command = m_command;
+      if (cmd.is_dict_key()) {
+        // torrent::Object tmp_command = cmd;
 
         // uint32_t flags = tmp_command.flags() & torrent::Object::mask_function;
         // tmp_command.unset_flags(torrent::Object::mask_function);
@@ -109,10 +113,10 @@
         // result = rpc::commands.call_command(tmp_command.as_dict_key().c_str(), tmp_command.as_dict_obj(),
         //                                     rpc::make_target(d1));
 
-        result = rpc::commands.call_command(m_command.as_dict_key().c_str(), m_command.as_dict_obj(), rpc::make_target(d1));
+        result = rpc::commands.call_command(cmd.as_dict_key().c_str(), cmd.as_dict_obj(), rpc::make_target(d1));
 
       } else {
-        result = rpc::parse_command_single(rpc::make_target(d1), m_command.as_string());
+        result = rpc::parse_command_single(rpc::make_target(d1), cmd.as_string());
       }
 
       switch (result.type()) {
@@ -136,6 +140,7 @@
   }
 
   const torrent::Object&       m_command;
+  const torrent::Object&       m_command2;
 };
 
 void
@@ -258,8 +263,8 @@
 void
 View::filter() {
   // Parition the list in two steps so we know which elements changed.
-  iterator splitVisible  = std::stable_partition(begin_visible(),  end_visible(),  view_downloads_filter(m_filter));
-  iterator splitFiltered = std::stable_partition(begin_filtered(), end_filtered(), view_downloads_filter(m_filter));
+  iterator splitVisible  = std::stable_partition(begin_visible(),  end_visible(),  view_downloads_filter(m_filter, m_tempFilter));
+  iterator splitFiltered = std::stable_partition(begin_filtered(), end_filtered(), view_downloads_filter(m_filter, m_tempFilter));
 
   base_type changed(splitVisible, splitFiltered);
   iterator splitChanged = changed.begin() + std::distance(splitVisible, end_visible());
@@ -298,7 +303,7 @@
   if (itr == base_type::end())
     throw torrent::internal_error("View::filter_download(...) could not find download.");
 
-  if (view_downloads_filter(m_filter)(download)) {
+  if (view_downloads_filter(m_filter, m_tempFilter)(download)) {
       
     if (itr >= end_visible()) {
       erase_internal(itr);
diff -ur rtorrent-0.9.6/src/core/view.h rtorrent-0.9.6-temp-filter/src/core/view.h
--- rtorrent-0.9.6/src/core/view.h	2015-09-03 20:03:30.000000000 +0100
+++ rtorrent-0.9.6-temp-filter/src/core/view.h	2020-03-10 12:22:05.704143733 +0000
@@ -124,6 +124,8 @@
 
   const torrent::Object& get_filter() const { return m_filter; }
   void                set_filter(const torrent::Object& s)        { m_filter = s; }
+  const torrent::Object& get_temp_filter() const { return m_tempFilter; }
+  void                set_temp_filter(const torrent::Object& s)        { m_tempFilter = s; }
   void                set_filter_on_event(const std::string& event);
 
   void                clear_filter_on();
@@ -172,6 +174,7 @@
   torrent::Object     m_sortCurrent;
 
   torrent::Object     m_filter;
+  torrent::Object     m_tempFilter; // Temporary view filter (eg: name based filter)
 
   torrent::Object     m_event_added;
   torrent::Object     m_event_removed;
diff -ur rtorrent-0.9.6/src/core/view_manager.cc rtorrent-0.9.6-temp-filter/src/core/view_manager.cc
--- rtorrent-0.9.6/src/core/view_manager.cc	2015-08-08 16:40:41.000000000 +0100
+++ rtorrent-0.9.6-temp-filter/src/core/view_manager.cc	2020-03-10 12:22:05.704143733 +0000
@@ -110,6 +110,14 @@
 }
 
 void
+ViewManager::set_temp_filter(const std::string& name, const torrent::Object& cmd) {
+  iterator viewItr = find_throw(name);
+
+  (*viewItr)->set_temp_filter(cmd);
+  (*viewItr)->filter();
+}
+
+void
 ViewManager::set_filter_on(const std::string& name, const filter_args& args) {
   iterator viewItr = find_throw(name);
 
diff -ur rtorrent-0.9.6/src/core/view_manager.h rtorrent-0.9.6-temp-filter/src/core/view_manager.h
--- rtorrent-0.9.6/src/core/view_manager.h	2015-08-08 16:40:41.000000000 +0100
+++ rtorrent-0.9.6-temp-filter/src/core/view_manager.h	2020-03-10 12:22:05.704143733 +0000
@@ -93,6 +93,7 @@
   void                set_sort_current(const std::string& name, const torrent::Object& cmd) { (*find_throw(name))->set_sort_current(cmd); }
 
   void                set_filter(const std::string& name, const torrent::Object& cmd);
+  void                set_temp_filter(const std::string& name, const torrent::Object& cmd);
   void                set_filter_on(const std::string& name, const filter_args& args);
 
   void                set_event_added(const std::string& name, const torrent::Object& cmd)   { (*find_throw(name))->set_event_added(cmd); }
diff -ur rtorrent-0.9.6/src/display/window_download_list.cc rtorrent-0.9.6-temp-filter/src/display/window_download_list.cc
--- rtorrent-0.9.6/src/display/window_download_list.cc	2015-09-03 20:03:30.000000000 +0100
+++ rtorrent-0.9.6-temp-filter/src/display/window_download_list.cc	2020-03-10 12:22:05.704143733 +0000
@@ -80,7 +80,7 @@
   if (m_view == NULL)
     return;
 
-  m_canvas->print(0, 0, "%s", ("[View: " + m_view->name() + "]").c_str());
+  m_canvas->print(0, 0, "%s", ("[View: " + m_view->name() + (m_view->get_temp_filter().is_empty() ? "" : " (filtered)") + "]").c_str());
 
   if (m_view->empty_visible() || m_canvas->width() < 5 || m_canvas->height() < 2)
     return;
diff -ur rtorrent-0.9.6/src/ui/download_list.cc rtorrent-0.9.6-temp-filter/src/ui/download_list.cc
--- rtorrent-0.9.6/src/ui/download_list.cc	2015-09-03 20:03:30.000000000 +0100
+++ rtorrent-0.9.6-temp-filter/src/ui/download_list.cc	2020-03-10 12:23:41.611779703 +0000
@@ -262,6 +262,15 @@
     title = "command";
     break;
 
+  case INPUT_FILTER:
+    // STARTED and STOPPED views are not allowed to being filtered: they are special
+    if (current_view()->name() == "started" || current_view()->name() == "stopped") {
+      control->core()->push_log_std("View '" + current_view()->name() + "' can't be filtered.");
+      return;
+    }
+    title = "filter";
+    break;
+
   default:
     throw torrent::internal_error("DownloadList::receive_view_input(...) Invalid input type.");
   }
@@ -325,6 +334,28 @@
                                 input->str());
       break;
 
+    case INPUT_FILTER:
+      if (input->str().empty()) {
+        if (rpc::call_command_value("ui.console.log.tempfilter"))
+          control->core()->push_log_std("Clear temporary filter on '" + current_view()->name() + "' view.");
+        current_view()->set_temp_filter(torrent::Object());
+        current_view()->filter();
+        current_view()->sort();
+      } else {
+        std::string pattern = input->str();
+        if (pattern.back() != '$')
+          pattern = pattern + ".*";
+        if (pattern.front() != '^')
+          pattern = ".*" + pattern;
+        std::transform(pattern.begin(), pattern.end(), pattern.begin(), ::tolower);
+        std::string tempFilter = "match={d.name=," + pattern + "}";
+        if (rpc::call_command_value("ui.console.log.tempfilter"))
+          control->core()->push_log_std("Temporary filter on '" + current_view()->name() + "' view: " + pattern);
+        current_view()->set_temp_filter(tempFilter);
+        current_view()->filter();
+      }
+      break;
+
     default:
       throw torrent::internal_error("DownloadList::receive_exit_input(...) Invalid input type.");
     }
@@ -346,6 +377,7 @@
   m_bindings[KEY_ENTER]     = std::tr1::bind(&DownloadList::receive_view_input, this, INPUT_LOAD_MODIFIED);
   m_bindings['\x0F']        = std::tr1::bind(&DownloadList::receive_view_input, this, INPUT_CHANGE_DIRECTORY);
   m_bindings['X' - '@']     = std::tr1::bind(&DownloadList::receive_view_input, this, INPUT_COMMAND);
+  m_bindings['F']           = std::tr1::bind(&DownloadList::receive_view_input, this, INPUT_FILTER);
 
   m_uiArray[DISPLAY_LOG]->bindings()[KEY_LEFT] =
     m_uiArray[DISPLAY_LOG]->bindings()['B' - '@'] =
diff -ur rtorrent-0.9.6/src/ui/download_list.h rtorrent-0.9.6-temp-filter/src/ui/download_list.h
--- rtorrent-0.9.6/src/ui/download_list.h	2015-09-03 20:03:30.000000000 +0100
+++ rtorrent-0.9.6-temp-filter/src/ui/download_list.h	2020-03-10 12:22:05.704143733 +0000
@@ -86,7 +86,8 @@
     INPUT_LOAD_DEFAULT,
     INPUT_LOAD_MODIFIED,
     INPUT_CHANGE_DIRECTORY,
-    INPUT_COMMAND
+    INPUT_COMMAND,
+    INPUT_FILTER
   } Input;
 
   DownloadList();
